react-components/src/components/NodeCacheProvider/AnnotationModelUtils.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { type CogniteClient, type Asset } from '@cognite/sdk/dist/src';
+import { uniqBy, chunk } from 'lodash';
+import { filterUndefined } from '../../utilities/filterUndefined';
+import { type RevealAnnotationModel } from './types';
+import { getAssetIdOrExternalIdFromAnnotation } from './utils';
+
+export async function fetchAnnotationAssets(
+  annotations: RevealAnnotationModel[],
+  sdk: CogniteClient
+): Promise<Map<number, Asset>> {
+  const annotationMapping = annotations.map((annotation) => {
+    const assetId = getAssetIdOrExternalIdFromAnnotation(annotation);
+    if (assetId === undefined) {
+      return undefined;
+    }
+    return {
+      annotationId: annotation.id,
+      assetId
+    };
+  });
+  const filteredAnnotationMapping = filterUndefined(annotationMapping);
+
+  const uniqueAnnotationMapping = uniqBy(filteredAnnotationMapping, 'assetId');
+  const assetIds = uniqueAnnotationMapping.map((mapping) => mapping.assetId);
+  const assets = await fetchAssetForAssetIds(assetIds, sdk);
+
+  const annotationIdToAssetMap = new Map<number, Asset>();
+  assets.forEach((asset) => {
+    filteredAnnotationMapping.forEach((mapping) => {
+      if (mapping.assetId === asset.id) {
+        annotationIdToAssetMap.set(mapping.annotationId, asset);
+      }
+    });
+  });
+  return annotationIdToAssetMap;
+}
+
+export async function fetchAssetForAssetIds(
+  assetIds: Array<string | number>,
+  sdk: CogniteClient
+): Promise<Asset[]> {
+  const assetsResult = await Promise.all(
+    chunk(assetIds, 1000).map(async (assetIdsChunck) => {
+      const retrievedAssets = await sdk.assets.retrieve(
```

Comment:
Just wondering - do we do a similar operation somewhere else in react-components? Could we factor this out as a reusable function?

Reply:
It is a re-usable function